File: python/src/main/python/Cone.py
class Cone:

    def prepare(self):
        print('Prepare cone...')

    def addIceCream(self, newIceCream):
        self.iceCream = newIceCream
        print('Adding  ' + self.iceCream.getFlavor() + ' to cone.')

    def serve(self):
        print('Serving...')

    def getIceCream(self):
        return self.iceCream

File: java/src/main/java/mx/iteso/factory/Cone.java
package mx.iteso.factory;

import mx.iteso.factory.icecreams.IceCream;

/** Cone class. */
public class Cone {

    /** Ice cream contained in cone. */
    private IceCream iceCream;

    /** Preparing alert method. */
    public void prepare() {
        System.out.println("Preparing ice cream");
    }

    /** Ice Cream adding method.
     * @param selectedIceCream the ice cream.
     */
    public void addIceCream(final IceCream selectedIceCream) {
        this.iceCream = selectedIceCream;
        System.out.println("Preparing " + iceCream.getFlavor() + " ice cream.");
    }

    /** Serving alert method. */
    public void serve() {
        System.out.println("Serving cone");
    }

    /** IceCream getter.
     * @return the ice cream contained in the cone.
     */
    public IceCream getIceCream() {
        return this.iceCream;
    }
}
